restrictedsessions/middleware.py

```diff
@@ -22,15 +22,25 @@ def process_request(self, request):
             return
 
         # Short circuit for option to require authed users
+        user = getattr(request, 'user', None)
         if getattr(settings, 'RESTRICTEDSESSIONS_AUTHED_ONLY', False):
-            user = getattr(request, 'user', None)
             # No logged in user -- ignore checks
             if not user or not hasattr(user, 'is_authenticated') or not user.is_authenticated():
                 return
 
         # Extract remote IP address for validation purposes
-        remote_addr_key = getattr(settings, 'RESTRICTEDSESSIONS_REMOTE_ADDR_KEY', 'REMOTE_ADDR')
-        remote_addr = request.META.get(remote_addr_key)
+        if getattr(settings, 'RESTRICTEDSESSIONS_REMOTE_ADDR_KEY', False):
+            remote_addr_key = getattr(settings, 'RESTRICTEDSESSIONS_REMOTE_ADDR_KEY')
+            remote_addr = request.META.get(remote_addr_key)
+        else:
+            if request.META.get('HTTP_X_FORWARDED_FOR', False):
```

Comment:
A silent usage of x-forwarded-for can be a security issue. There are situations where this may have been provided by the HTTP client, rather than set by the server. In those cases it can't be trusted.

How about simply looking at whether the value in `remote_addr` contains comma's, and if so split off the first element?